plugins/module_utils/network/sonic/argspec/qos_buffer/qos_buffer.py

```diff
@@ -0,0 +1,78 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2024 Dell Inc. or its subsidiaries. All Rights Reserved
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The arg spec for the sonic_qos_buffer module
+"""
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+class Qos_bufferArgs(object):  # pylint: disable=R0903
+    """The arg spec for the sonic_qos_buffer module
+    """
+
+    def __init__(self, **kwargs):
+        pass
+
+    argument_spec = {
+        'config': {
+            'options': {
+                'buffer_pools': {
+                    'elements': 'dict',
+                    'options': {
+                        'name': {
+                            'choices': ['ingress_lossless_pool'],
+                            'required': True,
+                            'type': 'str'
+                        },
+                        'xoff': {'type': 'int'}
+                    },
+                    'type': 'list'
+                },
+                'buffer_profiles': {
+                    'elements': 'dict',
+                    'mutually_exclusive': [['dynamic_threshold', 'static_threshold']],
+                    'options': {
+                        'dynamic_threshold': {'type': 'int'},
+                        'name': {'required': True, 'type': 'str'},
+                        'pause_threshold': {'type': 'int'},
+                        'pool': {
+                            'choices': ['ingress_lossless_pool', 'egress_lossless_pool', 'egress_lossy_pool'],
+                            'type': 'str'
+                        },
+                        'size': {'type': 'int'},
+                        'static_threshold': {'type': 'int'}
+                    },
+                    'type': 'list'
+                }
+            },
+            'type': 'dict'
+        },
+        'state': {
+            'choices': ['merged', 'deleted'],
```

Comment:
Any reason for not doing replaced and overridden?

Reply:
Just thought: should we add specific error messages for replaced and overridden? Otherwise when they are invoked, the message will be from ansible infra, like "operation can not be found".